File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode771.java
import java.util.HashSet;
import java.util.Scanner;
import java.util.Set;

public class Leetcode771 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter jewels string: ");
        String jewels = input.next();
        System.out.print("Enter stones string: ");
        String stones = input.next();
        input.close();
        System.out.println(numJewelsInStones(jewels, stones));
    }

    private static int numJewelsInStones(String jewels, String stones) {
        int count = 0;
        Set<Character> jewelSet = new HashSet<>();
        for (int i = 0; i < jewels.length(); i++) {
            jewelSet.add(jewels.charAt(i));
        }
        for (int i = 0; i < stones.length(); i++) {
            char currStone = stones.charAt(i);
            if (jewelSet.contains(currStone)) {
                count++;
            }
        }
        return count;
    }
}

File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode771.ts
function numJewelsInStones(jewels: string, stones: string): number {
    let count: number = 0;
    let set: Set<string> = new Set();
    for (let i: number = 0; i < jewels.length; i++) {
        set.add(jewels.charAt(i));
    }
    for (let i: number = 0; i < stones.length; i++) {
        if (set.has(stones.charAt(i))) {
            count++;
        }
    }
    return count;
}

let jewels: string = 'aA';
let stones: string = 'aAAbbbb';
console.log(numJewelsInStones(jewels, stones));

export { };